src/main/java/reposense/git/GitUtil.java

```diff
@@ -64,6 +69,19 @@ static String convertToFilterAuthorArgs(Author author) {
         return filterAuthorArgsBuilder.toString();
     }
 
+    /**
+     * Converts email to regex, adding optional regex groups for better matching.
+     */
+    private static String convertToEmailRegex(String email) {
+        String[] emails = email.split("@");
+        String authorEmailRegex = EMAIL_PREFIX_REGEX_GROUP;
+        authorEmailRegex += StringsUtil.replaceSpecialSymbols(emails[0], ".");
+        authorEmailRegex += EMAIL_SUFFIX_REGEX_GROUP;
+        authorEmailRegex += ".";
+        authorEmailRegex += StringsUtil.replaceSpecialSymbols(emails[1], ".");
```

Comment:
here seems to assume that the email will definitely have the "@", which might not be the case. 
Note that all these are user input and tend to have misconfiguration, internationally or not.

Reply:
Also, there isn't actually any need to `improve the matching`. The `StringsUtil.replaceSpecialSymbols` is just a security feature to ensure theres are no malicious injection.

Reply:
Oh no, I might have misunderstood the issue. I thought we were trying to improve matching of emails like `35282962+githubid@users.noreply.github.com`? What I did was to improve matching so that user can just specify `githubid@users.noreply.github.com` and `git log` will match emails with + operator prefixed or suffixed.

Reply:
> here seems to assume that the email will definitely have the "@", which might not be the case.
> Note that all these are user input and tend to have misconfiguration, internationally or n

We do a validation of the email when Author is created => `Author#validateEmails`, which checks that there is a "@" in the email.

Reply:
>Oh no, I might have misunderstood the issue. I thought we were trying to improve matching of emails like 35282962+githubid@users.noreply.github.com? What I did was to improve matching so that user can just specify githubid@users.noreply.github.com and git log will match emails with + operator prefixed or suffixed.

Yes you're right but there isn't a need to have a suffix, isnt it?

Reply:
Oh i see. I was trying to match emails like `mingyi+123@gmail.com`. I shall remove it.